src/app/qgsdxfexportdialog.cpp

```diff
@@ -797,6 +809,283 @@ void QgsDxfExportDialog::deselectDataDefinedBlocks()
+        mModel->setData( idx, element.attribute( QStringLiteral( "attribute-index" ), QStringLiteral( "-1" ) ) );
+
+        idx = mModel->index( idx.row(), ALLOW_DD_SYMBOL_BLOCKS_COL, idx.parent() );
+        mModel->setData( idx, element.attribute( QStringLiteral( "use_symbol_blocks" ), QStringLiteral( "0" ) ), Qt::CheckStateRole );
+
+        idx = mModel->index( idx.row(), MAXIMUM_DD_SYMBOL_BLOCKS_COL, idx.parent() );
+        mModel->setData( idx, element.attribute( QStringLiteral( "max_number_of_classes" ), QStringLiteral( "-1" ) ) );
+      }
+      else
+      {
+        QgsDebugMsgLevel( QStringLiteral( " Layer '%1' found in the DXF settings XML file, but not present in the project." ).arg( element.attribute( QStringLiteral( "name" ) ) ), 1 );
+      }
+    }
+  }
+
+  element = dxfElement.namedItem( QStringLiteral( "use_layer_title" ) ).toElement();
+  value = QgsXmlUtils::readVariant( element.firstChildElement() );
+  if ( !value.isNull() )
+    mLayerTitleAsName->setChecked( value == true );
+
+  element = dxfElement.namedItem( QStringLiteral( "use_map_extent" ) ).toElement();
+  value = QgsXmlUtils::readVariant( element.firstChildElement() );
+  if ( !value.isNull() )
+    mMapExtentCheckBox->setChecked( value == true );
+
+  element = dxfElement.namedItem( QStringLiteral( "force_2d" ) ).toElement();
+  value = QgsXmlUtils::readVariant( element.firstChildElement() );
+  if ( !value.isNull() )
+    mForce2d->setChecked( value == true );
+
+  element = dxfElement.namedItem( QStringLiteral( "mtext" ) ).toElement();
+  value = QgsXmlUtils::readVariant( element.firstChildElement() );
+  if ( !value.isNull() )
+    mMTextCheckBox->setChecked( value == true );
+
+  element = dxfElement.namedItem( QStringLiteral( "selected_features_only" ) ).toElement();
+  value = QgsXmlUtils::readVariant( element.firstChildElement() );
+  if ( !value.isNull() )
+    mSelectedFeaturesOnly->setChecked( value == true );
+
+  return true;
+}
+
+
+void QgsDxfExportDialog::saveSettingsToFile()
+{
+  QString outputFileName = QFileDialog::getSaveFileName( this, tr( "Save DXF Export settings as XML" ),
+                           QgsDxfExportDialog::settingsDxfLastSettingsDir->value(),
+                           tr( "XML file" ) + " (*.xml)" );
+  // return dialog focus on Mac
+  activateWindow();
+  raise();
+  if ( outputFileName.isEmpty() )
+  {
+    return;
+  }
+
+  //ensure the user never omitted the extension from the file name
+  if ( !outputFileName.endsWith( QStringLiteral( ".xml" ), Qt::CaseInsensitive ) )
+  {
+    outputFileName += QStringLiteral( ".xml" );
+  }
+
+  QString errorMessage;
+  QDomDocument domDocument;
+
+  saveSettingsToXML( domDocument );
+
+  const QFileInfo fileInfo( outputFileName );
+  const QFileInfo dirInfo( fileInfo.path() );  //excludes file name
+  if ( !dirInfo.isWritable() )
+  {
+    QMessageBox::information( this, tr( "Save DXF settings" ), tr( "The directory containing your dataset needs to be writable!" ) );
```

Comment:
```suggestion
    QMessageBox::information( this, tr( "Save DXF Settings" ), tr( "The directory containing your dataset needs to be writable!" ) );
```
Here and below. Missing "export" deliberate?